staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go

```diff
@@ -356,91 +371,31 @@ func destroyService(service Service) {
 	}
 }
 
-func getSocketName() string {
-	return fmt.Sprintf("unix:///@%s.sock", uuid.NewUUID())
-}
-
 // Test all those invalid configuration for KMS provider.
 func TestInvalidConfiguration(t *testing.T) {
 	t.Parallel()
 	// Start a test gRPC server.
-	f, err := startFakeKMSProvider(kmsapiVersion, getSocketName())
+	f, err := NewBase64Plugin(newTestSocket().path)
 	if err != nil {
 		t.Fatalf("failed to start test KMS provider server, error: %v", err)
 	}
-	defer f.Stop()
+	f.Start()
+	defer f.CleanUp()
 
 	invalidConfigs := []struct {
-		name       string
-		apiVersion string
-		endpoint   string
+		name     string
+		endpoint string
 	}{
-		{"emptyConfiguration", kmsapiVersion, ""},
-		{"invalidScheme", kmsapiVersion, "tcp://localhost:6060"},
+		{"emptyConfiguration", ""},
+		{"invalidScheme", "tcp://localhost:6060"},
 	}
 
 	for _, testCase := range invalidConfigs {
 		t.Run(testCase.name, func(t *testing.T) {
-			f.apiVersion = testCase.apiVersion
 			_, err := NewGRPCService(testCase.endpoint, 1*time.Second)
 			if err == nil {
 				t.Fatalf("should fail to create envelope service for %s.", testCase.name)
 			}
 		})
 	}
 }
-
-// Start the gRPC server that listens on unix socket.
-func startFakeKMSProvider(version, endpoint string) (*fakeKMSPlugin, error) {
-	sockFile, err := parseEndpoint(endpoint)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse endpoint:%q, error %v", endpoint, err)
-	}
-	listener, err := net.Listen(unixProtocol, sockFile)
-	if err != nil {
-		return nil, fmt.Errorf("failed to listen on the unix socket, error: %v", err)
-	}
-
-	s := grpc.NewServer()
-	f := &fakeKMSPlugin{apiVersion: version, server: s, sockFile: sockFile}
-	kmsapi.RegisterKeyManagementServiceServer(s, f)
-	go s.Serve(listener)
-	return f, nil
-}
-
-// Fake gRPC sever for remote KMS provider.
-// Use base64 to simulate encrypt and decrypt.
-type fakeKMSPlugin struct {
-	apiVersion string
-	server     *grpc.Server
-	sockFile   string
-}
-
-func (s *fakeKMSPlugin) Stop() {
-	// Stop the server
-	s.server.Stop()
-	// If this isn't a Linux abstract namespace socket, or if we're on a non-linux platform, clean up the socket file
```

Comment:
this breaks unit tests of this package on darwin

Reply:
Yes, I totally forgot about this.
Add the this logic into the CleanUp method.